lib/portlayer/network/context.go

```diff
@@ -1073,10 +1086,20 @@ func (c *Context) AddContainer(h *exec.Handle, options *AddContainerOptions) err
 		h.ExecConfig.Networks = make(map[string]*executor.NetworkEndpoint)
 	}
 
+	// If the scope type is external, name the nic as such.  We can only have
```

Comment:
Unless something changed since rebasing, i didn't observe it to be true.  I'll try on master and get back to you.

The problem is the network is configured in a different routine than setup firewall.  We could pass the scope type down along with the Port in the `tether.NetworkEndpoint`.  Is that what you're suggesting?  I didn't want to add redundant fields and I'm not that familiar with the API here so suggests are welcome.  This just seemed the most straight forward.